pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +40,28 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongLicenseSecretName
```

Comment:
This doesn't seem accurate :thinking: `b.license` is the actual license contents, but we're storing the secret name in it?

Now that we have `WithLicense(license string)` this will break that if `WithLicense` is run first.

Reply:
yes, assuming user needs install the secret before or using helm chart. in our case, we always install the default secret before using it in the build().

Reply:
Isn't `b.license` supposed to be the _actual content of the license_ though? `KongLicenseSecretName` isn't the content?

Reply:
f47f2681a07ca1cb9b759eb0011b81dd9eb886fb

Reply:
I'm a bit confused where we are at with this one, you marked this resolved and posted f47f268 but I don't see that commit as having any relevance to this: there still seems to be a bug here in that if you run the following:

```go
builder.WithLicense("<license>").WithEnterprise()
```

This will break the deployment because this line `b.license = KongLicenseSecretName` is not only overriding a value that was potentially already provided, but also seemingly overriding it with invalid data.

Reply:
Here is the implementation https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-7cc1dbd395c3c0ea6e62a7d5a3d4493f37054d7b7be23f198340cdd6a52c0be2R60. Here, the license serves as a helm charts parameters, indicating which license gateway will be used, the purpose is to use the data user provided, not see it has a a way to override the data. is it clear ?

Reply:
Sorry, this is not clear there still appears to be a problem.

Here's a patch that I applied to your branch including a unit test:

```diff
diff --git a/pkg/clusters/addons/kong/enterprise_test.go b/pkg/clusters/addons/kong/enterprise_test.go
new file mode 100644
index 0000000..7aa525f
--- /dev/null
+++ b/pkg/clusters/addons/kong/enterprise_test.go
@@ -0,0 +1,13 @@
+package kong
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEnterpriseLicense(t *testing.T) {
+	licenseTXT := "fake-test-license"
+	builder := NewBuilder().WithLicense(licenseTXT).WithEnterprise()
+	require.Equal(t, licenseTXT, builder.license)
+}
```

This unit test should (ostensibly, I suppose) pass, but it does not:

```shell
$ go test -v ./pkg/clusters/addons/kong/...
=== RUN   TestEnterpriseLicense
    enterprise_test.go:12: 
        	Error Trace:	enterprise_test.go:12
        	Error:      	Not equal: 
        	            	expected: "fake-test-license"
        	            	actual  : "kong-enterprise-license"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-fake-test-license
        	            	+kong-enterprise-license
        	Test:       	TestEnterpriseLicense
--- FAIL: TestEnterpriseLicense (0.00s)
FAIL
FAIL	github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong	0.007s
FAIL
```

This demonstrates the problem, and ultimately the entirely problem would be relieved if you took the license argument in the enterprise method (e.g. `func (b *Builder) WithEnterprise(license string) *Builder {`) as originally suggested, because then there's no order of operations to be concerned with when configuring the license.